src/components/select/multi-select/multi-select.component.js

```diff
@@ -111,6 +116,34 @@ const MultiSelect = React.forwardRef(
       [name, id]
     );
 
+    /* generic value update function which can be used for both controlled and uncontrolled
+     * components, both with and without onChange.
+     * - if onChange is provided, calls that with the updated value, provided it has changed in length
+     * (ie if an option has been selected or deselected)
+     * - if the component is uncontrolled, call setSelectedValue with the provided update function
+     * - if the component is controlled, do not change selectedValue and instead directly call
+     * the update function and pass its result to onChange
+     * - ensure that in every circumstance, onChange and the update function are only called at most
+     * once, in case they have side effects */
+    const setValue = useCallback(
```

Comment:
I thought it would be good to do this as it's really equivalent to a state update function - but you're right that it might be confusing because it strictly isn't one. I'll change it.

Reply:
I don't mind the approach but could see how the naming might be a little confusing even though it is replicating the `setState` functionality, something like `updateValue` might work better  but I'm happy to let you decide on that